### PR TITLE
[Script] fix AuthIT

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**

### CODE REVIEW

File: daml-script/test/src/test-utils/com/daml/lf/engine/script/test/CantonFixture.scala [hunk 1]
```diff
@@ -83,6 +133,11 @@ trait CantonFixture
             case ScriptTimeMode.Static => (Some("monotonic-time"), Some("sim-clock"))
             case ScriptTimeMode.WallClock => (None, None)
           }
+          val authConfig = authSecret.fold("")(secret => s"""auth-services = [{
+               |          type = unsafe-jwt-hmac-256
+               |          secret = "${toJson(secret)}"
+               |        }]
+               |""".stripMargin)
```
Comment (remyhaemmerle-da) [inline daml-script/test/src/test-utils/com/daml/lf/engine/script/test/CantonFixture.scala:140]: Here we setup the authentication in Canton

File: daml-script/test/src/test-utils/com/daml/lf/engine/script/test/AbstractScriptTest.scala [hunk 2]
```diff
@@ -46,28 +40,6 @@ object AbstractScriptTest {
       values = ArrayList(a, b),
     )
 
-  val appId = ApplicationId("daml-script-test")
-  private val secret = "secret"
-
-  def getToken(parties: List[String], admin: Boolean): String = {
```
Comment (remyhaemmerle-da) [inline daml-script/test/src/test-utils/com/daml/lf/engine/script/test/AbstractScriptTest.scala:52]: This is moved in CantonFixture and base on StandardDamlJWTPayload

File: daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/AuthIT.scala [hunk 3]
```diff
@@ -1,74 +1,77 @@
...
+  lazy val secret = "secret"
+
+  override protected lazy val authSecret = Some(secret)
+  override protected lazy val darFiles: List[File] = List(stableDarPath)
+  override protected lazy val devMode: Boolean = false
+  override protected lazy val nParticipants: Int = 1
+  override protected lazy val timeMode = ScriptTimeMode.WallClock
+  override protected lazy val tlsEnable: Boolean = false
+
   "Daml Script against authorized ledger" can {
```
Comment (remyhaemmerle-da) [inline daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/AuthIT.scala:36]: We change the test to make authentification by user instead parties, as users is the proper way to authenticate.

File: daml-script/test/src/test-utils/com/daml/lf/engine/script/test/CantonFixture.scala [hunk 4]
```diff
@@ -26,27 +28,74 @@ import scala.concurrent.duration.DurationInt
...
   protected def devMode: Boolean
+  protected def nParticipants: Int
   protected def timeMode: ScriptTimeMode
   protected def tlsEnable: Boolean
 
+  // This flag setup some behavior to ease debugging tests.
+  //  If `true`
+  //   - temporary file are not deleted
+  //   - some debug info are logged.
+  private val debugMode = false
```
[the remark this answers is not in the document]
Comment (carlpulley-da) [reply daml-script/test/src/test-utils/com/daml/lf/engine/script/test/CantonFixture.scala:68]: Would `cantonFixtureDebugMode` be a better name here? Simply wondering if `debugMode` is too generic?

File: daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/AuthIT.scala [hunk 5]
```diff
@@ -1,74 +1,77 @@
...
-          value = Some(
-            Value.ValueRecord(
-              None,
-              ImmArray(
-                None -> Value.ValueParty(Party.assertFromString(parties.head)),
-                None -> Value.ValueParty(Party.assertFromString(parties.tail.head)),
-              ),
-            )
+          adminClient <- defaultLedgerClient(getToken(adminUserId))
+          userId = Ref.UserId.assertFromString(CantonFixture.freshName("user"))
```
Comment (akrmn) [inline daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/AuthIT.scala:41]: I think you meant to have this instead

```suggestion
          userId = Ref.UserId.assertFromString(CantonFixture.freshUserId())
```

**Review by akrmn — APPROVED** [review]
LGTM, thank you @remyhaemmerle-da!

File: daml-script/test/src/test-utils/com/daml/lf/engine/script/test/CantonFixture.scala (hunk 4 above)
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply daml-script/test/src/test-utils/com/daml/lf/engine/script/test/CantonFixture.scala]: 7019c4c743c1fa29702cfe1a8c94823d0babc7cb

File: daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/AuthIT.scala (hunk 5 above)
Comment (remyhaemmerle-da) [reply daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/AuthIT.scala]: 7019c4c743c1fa29702cfe1a8c94823d0babc7cb

**Pull request merged**